javatests/arcs/core/data/TypeUtilsTest.kt

```diff
@@ -46,16 +49,39 @@ class TypeUtilsTest {
     }
   }
 
+  @Test
+  fun mapFieldTypeToInferredType_nullable() {
+    assertThat(
+      mapFieldType(FieldType.Long.nullable())
+    ).isEqualTo(
+      InferredType.UnionType(setOf(
+        InferredType.Primitive.LongType,
+        InferredType.Primitive.NullType
+      ))
+    )
+  }
+
+  @Test
+  fun nullableValueSupportDisabled_mapFieldTypeToInferredType_nullable() {
+    // Ensures that disabling nullable support falls back to non-nullable values.
+    BuildFlags.NULLABLE_VALUE_SUPPORT = false
```

Comment:
Worth adding one more test with the flag disabled that tries to construct `NullableOf` directly (and then `assertFailsWith`).

Reply:
Never mind, I see you've done that in SchemaFieldsTest.kt, which is the right place for it.